lib/throng.js

```diff
@@ -47,10 +48,17 @@ module.exports = async function throng(options, legacy) {
     }
   }
 
-  function revive() {
+  // We cannot rely on every exited worker sending `disconnect`,
+  // so listen to both `disconnect` and `exit` and track deaths
+  // to avoid duplicate calls to revive().
+  const deaths = []
```

Comment:
Is this an always increasing array? This doesn't seem right to me. We need a process to clean this up eventually. Maybe a timer, maybe weakmap?